docs/multichain.md

```diff
@@ -2,20 +2,190 @@
+Simperby can control various Dapps (Decentralized Applications) deployed on
+various existing chains, such as Ethereum. This could include a treasury for the
+organization, a token that the organization issues, or any other application
+that the organization wants to build and deploy. This multichain communication
+is performed in a **trustless and verifiable manner**, with reasonable cost and
+efficiency.
 
-## Consensus & Governance
+## Light Client
 
-TODO
+A light client is a special optimization of a blockchain node that is capable of
+verifying transactions or state entries without having the full blockchain
+state. This is also known as Simplified Payment Verification (SPV) in Bitcoin.
+
+The key idea of a light client is that a blockchain is still **cryptographically
+meaningful even if we take only the headers** of the chain (note that the block
+body contains transactions).
+
+In Tendermint and Vetomint, the light client has a "last trusted header" as its
+state, and it only accepts a new header if it is actually finalized by the
+consensus, as the next block to the last trusted header. **There is no authority
+or trust in the header provider** as there is always a cryptographic proof for the
+finalization, the pre-commits from over 2/3 of the validators. If both the
+content of the header and the accompanying pre-commits are valid, the light
+client can update its state to the new header. Note that to verify the
+pre-commits, the light client also must know the eligible validator set for the
+new block. Fortunately, a block header contains either the validator set itself
+(Simperby's way) or the hash of it (most other blockchains do this way because they
+produce blocks frequently, so reducing the header size is important), so verification can
+be trivially handled.
+
+Now we know that a light client can track the canonical chain of headers only by
+constantly accepting the headers and proof in a trustless and verifiable way.
+Since a block header contains the Merkle root of the state and transactions, the
+**light client can verify the inclusion of any state entry or transaction with
+its Merkle proof**, at the given height. The light client may store all previous
+headers that it has accepted so far to verify the inclusion from past blocks.
 
 ## Trustless Message Delivery
 
+A light client takes blockchain headers and their proofs as the only input to
+update its state. In other words, the light client can stay in sync with the
+blockchain without accessing the network if there are constant header updates
+from whoever submits valid ones.
+
+Thanks to this simple but powerful property, it is possible to embed a light
+client as a smart contract. This enables **trustless message delivery between
+two blockchains**, which is the key to multichain interoperability.
+
+Suppose there are two chains called A and B. If the A chain has a light client
+of the B Chain deployed on it, whatever is finalized in the B chain can be
```

Comment:
```suggestion
of the B Chain deployed on A chain, whatever finalized on the B chain can be
```

Maybe in and on both works with slightly different meanings. I want to ask Chatgpt but not possible at the moment.